test/functional/shared/rptest.go

```diff
@@ -303,128 +305,137 @@ func (ct RPTest) Test(t *testing.T) {
 	success := true
 	for i, step := range ct.Steps {
 		success = t.Run(step.Executor.GetDescription(), func(t *testing.T) {
-			defer ct.CleanUpExtensionResources(step.K8sOutputResources)
 			if !success {
 				t.Skip("skipping due to previous step failure")
 				return
 			}
 
-			t.Logf("running step %d of %d: %s", i, len(ct.Steps), step.Executor.GetDescription())
-			step.Executor.Execute(ctx, t, ct.Options.TestOptions)
-			t.Logf("finished running step %d of %d: %s", i, len(ct.Steps), step.Executor.GetDescription())
-
-			if step.SkipKubernetesOutputResourceValidation {
-				t.Logf("skipping validation of resources...")
-			} else if step.RPResources == nil || len(step.RPResources.Resources) == 0 {
-				require.Fail(t, "no resource set was specified and SkipKubernetesOutputResourceValidation == false, either specify a resource set or set SkipResourceValidation = true ")
-			} else {
-				// Validate that all expected output resources are created
-				t.Logf("validating output resources for %s", step.Executor.GetDescription())
-				validation.ValidateRPResources(ctx, t, step.RPResources, ct.Options.ManagementClient)
-				t.Logf("finished validating output resources for %s", step.Executor.GetDescription())
-			}
+			retry.Run(t, func(r *retry.R) {
```

Comment:
@willdavsmith This is the test util from hashicorp consul to enable retry.  cc/ @ytimocin